app/scripts/controllers/exportImport.js

```diff
@@ -199,111 +202,44 @@ angular.module('openhimConsoleApp')
+      $scope.importStatus = 'done';
+      $scope.importResult = data;
     };
 
     // function to run import file
-    $scope.runImportFile = function(data){
-
-      // set counter variables
-      $scope.importFail = 0;
-      $scope.importSuccess = 0;
-      $scope.failedImports = [];
-      $scope.importProgressStatus = 0;
-      $scope.importProgressType = '';
+    $scope.runImportFile = function(data) {
+      $scope.importStatus = 'progress';
+      $scope.updateFlag = false;
 
-      // convert to json object
       data = JSON.parse(data);
-
-      var totalRecords = 0;
-      if (data.Clients) {
-        totalRecords += data.Clients.length;
-      }
-      if (data.Channels) {
-        totalRecords += data.Channels.length;
-      }
-      if (data.ContactGroups) {
-        totalRecords += data.ContactGroups.length;
-      }
-      if (data.Users) {
-        totalRecords += data.Users.length;
-      }
-      if (data.Mediators) {
-        totalRecords += data.Mediators.length;
-      }
-
-      var doneItems = 0;
-
-      //loop through each collection in object
-      angular.forEach(data, function(modelRecords, model) {
-
-        // loop through each record
-        angular.forEach(modelRecords, function(value) {
-
-          var record;
-          var insertParams = {};
-
-          // check model to determine which API to call
-          switch(model) {
-            case 'Clients':
-              record = new Api.Clients( value );
-              break;
-            case 'Channels':
-              record = new Api.Channels( value );
-              break;
-            case 'ContactGroups':
-              record = new Api.ContactGroups( value );
-              break;
-            case 'Mediators':
-              if ( value._id ){
-                delete value._id;
-              }
-              record = new Api.Mediators( value );
-              insertParams.urn = '';
-              break;
-            case 'Users':
-              record = new Api.Users( value );
-              insertParams.email = '';
-              break;
-            default:
-              // no default yet
-          }
-
-          // if record has _id then do update
-          if ( record._id ){
-            record.$update(function(){
-              importSuccess();
-            }, function(err){
-              importFail(model, value, err);
-            });
-          }else{
-            record.$save(insertParams, function(){
-              importSuccess();
-            }, function(err){
-              importFail(model, value, err);
-            });
-          }
-
-          doneItems++;
-          $scope.importProgressStatus = Math.floor( doneItems / totalRecords );
-
-          // update progress bar too 100%
-          if( doneItems === totalRecords ){
-            $scope.importProgressStatus = 100;
-            $scope.importProgressType = 'success';
-          }
-
+      
+      // If record has _id then do update instead of insert
```

Comment:
Ok, but what we would loose is the ability to update existing collection unless you look them up by their unique properties. Are you sure this is possible for each collection type? If so, then maybe we can simplify an only have 1 export option which doesn't include the `_id`s.

cc/ @hnnesv for your thoughts.